src/object/cli_ec.c

```diff
@@ -367,25 +475,46 @@ ec_get_tgt_set(daos_iod_t *iods, unsigned int nr, struct daos_oclass_attr *oca,
 						iods[i].iod_size + rs - 1;
 
 			/* No partial-parity updates, so this function won't be
-			 * called if parity is present.
+			 * called if parity is present in a update request.
+			 * For fetch (!parity_include), the code allows parity
+			 * segments to be requested (and handles them
+			 * separately).
+			 */
+			if (PARITY_INDICATOR & rs) {
+				/* This allows selecting a parity target
+				 * for fetch. If combined with regualar
+				 * data extents, parity ranges must come
+				 * first in the the recx array.
+				 */
+				D_ASSERT(!parity_include);
+				ss = p * len;
+				p_offset = 0;
+				full = (1UL << p) - 1;
```

Comment:
is "full" intentionally overwritten at here? Because it's not initialized for the next iod

Reply:
This was put in to support a request that was trying to fetch parity only.  I've changed it so the fetch of parity is allowed, but the fetch "full' criteria is all of the data targets.  Full is just a way to end the target selection once all the eligible targets are selected (i.e, the set is full).